pkg/kubelet/pod_workers.go

```diff
@@ -1161,9 +1164,13 @@ func (p *podWorkers) removeTerminatedWorker(uid types.UID) {
 		return
 	}
 
-	if startedUID, started := p.startedStaticPodsByFullname[status.fullname]; started && startedUID != uid {
-		klog.V(4).InfoS("Pod cannot start yet but is no longer known to the kubelet, finish it", "podUID", uid)
-		status.finished = true
+	_, waiting := p.waitingToStartStaticPodsByFullname[status.fullname]
+	startedUID, started := p.startedStaticPodsByFullname[status.fullname]
+	if waiting || started {
```

Comment:
This condition block implies "if static pod" ...

I did it because there is only uid parameter in this function.
Do we have to add isStaticPod to podSyncStatus struct?

Reply:
I'm asking "why do you need to process this here" - i.e. describe the problem this code block is trying to solve.

Reply:
The whole block solves the problem that we could not remove a pod worker of the static pod that has been terminated before start.

We have to clean up the undesired pod worker that does not start yet or there is another static pod with the same name.

And we have to check waiting here because there is no way to check whether it is a static pod.

What I wanted to do is that

```go
if staticPod {
    if !started || (started && startedUID != uid) {
        klog.V(4).InfoS("Pod does not start yet but is no longer known to the kubelet, finish it", "podUID", uid)
	status.finished = true
    }
}

Reply:
Ok, so you're trying to stop a worker during the cleanup phase.  Yes, this is definitely broken.

The core issue here is that the cleanup phase (syncing known pods) for an unstarted pod is not the same as when a worker itself completes.

I think the current approach may not be correct because this is the opposite flow from the normal worker termination (where the worker is who sets .finished).  Thinking about the problem.

Reply:
https://github.com/kubernetes/kubernetes/pull/107900/files#diff-c6d21f5481c391a3f32bcfddf45c9c20132b2876e8be9690289474b76b71d809R868 i think is going to be easier to deal with long run - the only thing that closes a loop is the loop itself.